src/main/java/org/tron/program/Benchmark.java

```diff
@@ -0,0 +1,263 @@
+    String libraryAddressPair = null;
+
+    TVMResult result = VMTimeBenchmarkUtils
+        .deployContractAndReturnTVMResult(contractName, address, ABI, code,
+            value,
+            feeLimit, consumeUserResourcePercent, libraryAddressPair,
+            dbManager, null);
+
+    byte[] contractAddress = result.getContractAddress();
+
+    long totalDuration = 0;
+    int repeatCount = 12;
+
+    for (int i = 0; i < repeatCount; i++) {
+      long curDuration = triggerContractAndReturnDuration(contractAddress, feeLimit);
+      // System.out.println(String.format("count: %d, duration: %d", i, curDuration));
+      if (i >= 2) {
+        totalDuration += curDuration;
+      }
+      Thread.sleep(10);
+    }
+    long avgDuration = totalDuration / (repeatCount - 2);
+    // System.out.println(String.format("avg duration: %d", avgDuration));
+
+    double defaultMinTimeRatio = 0.0;
+    double defaultMaxTimeRatio = 5.0;
+    long timeBenchmark = 50L;
+    return new TimeBenchmarkResult(defaultMinTimeRatio * avgDuration / timeBenchmark,
+        defaultMaxTimeRatio * avgDuration / timeBenchmark);
+
+  }
+
+  public static long triggerContractAndReturnDuration(byte[] contractAddress, long feeLimit)
+      throws ContractExeException, ReceiptCheckErrException, VMIllegalException, ContractValidateException {
+    /* ====================================================================== */
+    String params = "0000000000000000000000000000000000000000000000000000000000004e20" +
+        "00000000000000000000000000000000000000000000000000000000000007d0" +
+        "00000000000000000000000000000000000000000000000000000000000000c8";
+    byte[] triggerData = VMTimeBenchmarkUtils
+        .parseABI("timeBenchmark(uint256,uint256,uint256)", params);
+    TVMResult result = VMTimeBenchmarkUtils
+        .triggerContractAndReturnTVMResult(Hex.decode(OWNER_ADDRESS),
+            contractAddress, triggerData, 0, feeLimit, dbManager, null);
+
+    return result.getDuration();
+  }
+
+  public static TimeBenchmarkResult getTimeRatio()
+      throws ContractExeException, ReceiptCheckErrException, VMIllegalException, ContractValidateException, InterruptedException {
+
+    TimeBenchmarkResult result = timeBenchmark();
+
+    return result;
+  }
+
+  public static long getMem() {
+
+    com.sun.management.OperatingSystemMXBean os = (com.sun.management.OperatingSystemMXBean)
+        java.lang.management.ManagementFactory.getOperatingSystemMXBean();
+    long physicalMemorySize = os.getTotalPhysicalMemorySize();
+
+    double mem = physicalMemorySize / 1024 / 1024 * 0.8;
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Cast one of the operands of this integer division to a "double". ([squid:S2184](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2184))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1546&issues=AWXsCYUK2IDoRcZ5ppJN&open=AWXsCYUK2IDoRcZ5ppJN)

[ISSUE_KEY]: AWXsCYUK2IDoRcZ5ppJN